Mage/src/main/java/mage/abilities/dynamicvalue/DynamicValue.java

```diff
@@ -17,7 +19,39 @@ public interface DynamicValue extends Serializable, Copyable<DynamicValue> {
 
     DynamicValue copy();
 
-    String getMessage();
+    // TODO: this will be removed once all DynamicValue implementations have been converted
+    // WARNING: You must override at least one getMessage in a DynamicValue implementation, otherwise these will loop infinitely.
```

Comment:
is it really necessary for the methods to call each other? I don't love that

Reply:
Addressed in https://github.com/magefree/mage/pull/12596#issuecomment-2325102617